File: JavaScript/7js 1/evaporator.js
/*
|--------------------------------------------------------------------------
| Deodorant Evaporator
|--------------------------------------------------------------------------
|
*/

function evaporator(content, evap_per_day, threshold) {
   let res = 0;
   let x = 100;
   while (x >= threshold) {
      x -= (x * evap_per_day) / 100;
      res++;
   }
   return res;
}


function evaporator(content, evap_per_day, threshold) {
   return Math.ceil(Math.log(threshold / 100) / Math.log((100 - evap_per_day) / 100));
}


function evaporator(content, evap_per_day, threshold) {
   function recurse(pctLeft = 100, counter = 0) {
      return pctLeft <= threshold
         ? counter
         : recurse(pctLeft * (1 - evap_per_day / 100), ++counter);
   }
   return recurse();
}


console.log(evaporator(10, 10, 10), 22);

File: Python/7py 1/evaporator.py
"""
|--------------------------------------------------------------------------
| Deodorant Evaporator
|--------------------------------------------------------------------------
|
"""

def evaporator(content, evap_per_day, threshold):
   res = 0
   x = 100
   while x >= threshold:
      x -= (x * evap_per_day) / 100
      res += 1
   return res


from math import ceil, log
def evaporator(c, e, t):
   return ceil(log(t/100, 1-(e/100)))


def evaporator(content, evap_per_day, threshold):
   days = 0
   while (((1-evap_per_day/100)**days) > (threshold/100)):
      days += 1
   return days


print(evaporator(10, 10, 10), 22)